### PR TITLE
Feat : add Registration form and middlewares for controls

### PR DESCRIPTION
**Pull request by PierreA44**

### CODE REVIEW

File: backend/src/middlewares/userValidation.js [hunk 1]
```diff
@@ -0,0 +1,43 @@
+const { z } = require("zod");
+
+const userSchema = z.object({
+  firstname: z
+    .string()
+    .min(2, { message: "Firstname must contain at least 2 characters" })
```
Comment (ioayoub) [inline backend/src/middlewares/userValidation.js:6]: When you need a specific message depending on value prefer using the syntax :
min({value :2, message: "your message"}). Easier to read, so easier to maintain

File: backend/database/schema.sql [hunk 2]
```diff
@@ -122,9 +122,9 @@ CREATE TABLE user (
     mail varchar(255)  NOT NULL,
     password varchar(255)  NOT NULL,
     week_time_kitchen int  NOT NULL,
-    sexe bool  NOT NULL,
+    gender VARCHAR(50)  NOT NULL,
```
Comment (ioayoub) [inline backend/database/schema.sql:125]: You are not anymore allowed to store genders because of GPDR (Rgpd) rules.

**Review by ioayoub — CHANGES_REQUESTED** [review]

**Review by Mewlyscin — APPROVED** [review]

File: backend/database/schema.sql [hunk 3]
```diff
@@ -121,9 +121,8 @@ CREATE TABLE user (
     mail varchar(255)  NOT NULL,
     password varchar(255)  NOT NULL,
     week_time_kitchen int  NOT NULL,
-    sexe bool  NOT NULL,
     weight int  NOT NULL,
-    registration_date date  NOT NULL,
+    registration_date DATETIME NOT NULL DEFAULT CURRENT_TIMESTAMP,
```
Comment (ioayoub) [inline backend/database/schema.sql:125]: Use registration_date DATETIME NOT NULL NOW() here

File: backend/src/models/RoleManager.js [hunk 4]
```diff
@@ -31,7 +31,7 @@ class RoleManager extends AbstractManager {
 
   async update(name, id) {
```
Comment (ioayoub) [inline backend/src/models/RoleManager.js:32]: name or type ? 🤔

File: backend/src/models/UserManager.js [hunk 5]
```diff
@@ -46,16 +45,16 @@ class UserManager extends AbstractManager {
...
         user.firstname,
         user.lastname,
-        user.email,
+        user.mail,
         user.pseudo,
         user.password,
         user.sexe,
-        user.weight,
         user.week_time_kitchen,
+        user.registration_date,
```
Comment (ioayoub) [inline backend/src/models/UserManager.js:57]: id is missing here

**Review by ioayoub — CHANGES_REQUESTED** [review]

File: backend/src/models/RoleManager.js (hunk 4 above)
Comment (PierreA44) [reply backend/src/models/RoleManager.js:32]: thank you for the look!

File: backend/src/models/UserManager.js (hunk 5 above)
Comment (PierreA44) [reply backend/src/models/UserManager.js:57]: i don't underdstand very well...

File: backend/database/schema.sql (hunk 3 above)
Comment (PierreA44) [reply backend/database/schema.sql:125]: Ok, it's good with that !

File: backend/src/models/UserManager.js (hunk 5 above)
Comment (PierreA44) [reply backend/src/models/UserManager.js:57]: it's good I finally understood ^^

**Review by ioayoub — APPROVED** [review]

**Review by akornne — APPROVED** [review]

**Review by bayram-69 — APPROVED** [review]

**Review by QuentinVerwaerde — APPROVED** [review]

**Pull request merged**